### PR TITLE
NOISSUE - Update Export

### PR DESCRIPTION
**Pull request by SammyOina**
### What does this do?
Make export compatible to mainflux 0.13

### Which issue(s) does this PR fix/relate to?
No issue

### List any changes that modify/break current functionality
None

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
Yes

### CODE REVIEW

**Review by drasko — APPROVED** [review]
LGTM

**Comment by rodneyosodo** [comment]
Can you try and see if this works with `RabbitMQ` configured as the internal message broker for mainflux?

**Comment by SammyOina** [comment]
> Can you try and see if this works with `RabbitMQ` configured as the internal message broker for mainflux?

this is best left to another PR

**Review by rodneyosodo — APPROVED** [review]
LGTM

**Review by rodneyosodo — CHANGES_REQUESTED** [review]
Add health endpoint

**Comment by rodneyosodo** [comment]
Similar situation as from here https://github.com/mainflux/agent/pull/53#discussion_r1259597936

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: CI
+on:
+  push:
+    branches: [main]
```
Comment (rodneyosodo) [inline .github/workflows/ci.yml:4]:
```suggestion
    branches: [master]
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  pull_request:
+    branches: [main]
```
Comment (rodneyosodo) [inline .github/workflows/ci.yml:6]:
```suggestion
    branches: [master]
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: pkg/export/api/transport.go [hunk 3]
```diff
@@ -1,36 +1,25 @@
...
-	fullMatch                 = []string{"state", "external_id", "mainflux_id", "mainflux_key"}
-	partialMatch              = []string{"name"}
-)
-
 // MakeHandler returns a HTTP API handler with version and metrics.
 func MakeHandler(svc export.Service) http.Handler {
 	r := bone.New()
 	r.GetFunc("/version", export.Version())
 	r.Handle("/metrics", promhttp.Handler())
+	r.GetFunc("/health", mainflux.Health("export"))
```
Comment (rodneyosodo) [inline pkg/export/api/transport.go:23]: No need to have `/version` as `/health` also implements version

Comment (rodneyosodo) [reply pkg/export/api/transport.go:22]: Update build command also

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: docker/.env [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+MF_EXPORT_CONFIG_FILE=config.toml
+MF_EXPORT_MQTT_HOST=tcp://mqtt-adapter:1883
+MF_EXPORT_MQTT_USERNAME="77d74527-7457-4dc2-9b36-01f01ce62726"
+MF_EXPORT_MQTT_PASSWORD="0f6bc453-4e5f-40f6-bc94-de8238b60c19"
+MF_EXPORT_MQTT_CHANNEL=05ff7d0b-62be-433a-9eff-cd963b4c6b5d
+MF_EXPORT_MQTT_SKIP_TLS=true
+MF_EXPORT_MQTT_MTLS=false
+MF_EXPORT_MQTT_CA=""
+MF_EXPORT_MQTT_CLIENT_CERT=""
+MF_EXPORT_MQTT_CLIENT_PK=""
```
Comment (rodneyosodo) [inline docker/.env:16]: The `MF_EXPORT_CHANNELS_CONFIG` variable is not set.

File: scripts/configure.sh [hunk 5]
```diff
@@ -11,18 +11,18 @@ EXPORT_CONFIG_FILE='./configs/export-config.toml'
 
 if [ "$MTLS" == true ]; then
     token=`curl -s -S -X POST https://${MAINFLUX_HOST}/tokens -d "{\"email\":\"${MAINFLUX_USER_EMAIL}\",\"password\":\"${MAINFLUX_USER_PASSWORD}\"}" -H 'Content-Type: application/json' |jq -r .token`
```
Comment (rodneyosodo) [inline scripts/configure.sh:13]: This has not been updated

File: scripts/configure.sh [hunk 6]
```diff
@@ -11,18 +11,18 @@ EXPORT_CONFIG_FILE='./configs/export-config.toml'
...
+    bootstrapResponse=`curl -s -S -X GET https://${MAINFLUX_HOST}/bootstrap/things/bootstrap/${EXTERNAL_ID} -H "Authorization: Thing ${EXTERNAL_KEY}" -H 'Content-Type: application/json'`
     thingID=`echo "${bootstrapResponse}" | jq -r .mainflux_id`
-    exportChannel=`curl -s -S -X GET https://${MAINFLUX_HOST}/things/${thingID}  -H 'Content-Type: application/json' -H "Authorization: ${token}" | jq -r .metadata.export_channel_id`
+    exportChannel=`curl -s -S -X GET https://${MAINFLUX_HOST}/things/${thingID}  -H 'Content-Type: application/json' -H "Authorization: Bearer ${token}" | jq -r .metadata.export_channel_id`
     clientKey=`echo "${bootstrapResponse}" | jq -r .client_key`
     clientCert=$(echo "${bootstrapResponse}" | jq -r .client_cert)
     caCert=`echo "${bootstrapResponse}" | jq -r .ca_cert`
     mqttHost="tcps://${MAINFLUX_HOST}:8883"
 else
     token=`curl -s --insecure -S -X POST https://${MAINFLUX_HOST}/tokens -d "{\"email\":\"${MAINFLUX_USER_EMAIL}\",\"password\":\"${MAINFLUX_USER_PASSWORD}\"}" -H 'Content-Type: application/json' |jq -r .token`
```
Comment (rodneyosodo) [inline scripts/configure.sh:22]: same

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: go.mod [hunk 7]
```diff
@@ -1,15 +1,37 @@
 module github.com/mainflux/export
 
-go 1.14
+go 1.20
 
 require (
-	github.com/eclipse/paho.mqtt.golang v1.2.0
-	github.com/go-kit/kit v0.10.0
-	github.com/go-redis/redis v6.15.8+incompatible
+	github.com/eclipse/paho.mqtt.golang v1.4.2
```
Comment (dborovcanin) [inline go.mod:6]: PLease check for the latest version.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: go.mod (hunk 7 above)
Comment (SammyOina) [reply go.mod:6]: this seems to  be the latest https://github.com/eclipse/paho.mqtt.golang/releases

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**